### PR TITLE
aws eks terraform playbooks

### PR DESCRIPTION
**Pull request by nonsense**
This PR is adding Terraform playbook for creating an AWS EKS cluster.

We haven't decided which orchestrator or cloud we will use for the Testground MVP, but I think it is good to keep our options open and commit the playbooks / setups we are evaluating.

### CODE REVIEW

**Comment by nonsense** [comment]
I think it is important for now to commit the setup/playbooks for various infra we evaluate, so that we have a point of reference in the future on how we tested a given provider. I wouldn't be surprised if 1 year from now there are alternative approaches and more options on how to setup AWS EKS, and it might become a better option than let's say DO.

**Comment by jimpick** [comment]
Interesting. Are we going to make a runner for k8s? As long as the networking is configured, we can probably run jobs manually without a runner (like what I did with Docker Swarm before we had the runner working).

**Comment by nonsense** [comment]
> Interesting. Are we going to make a runner for k8s? As long as the networking is configured, we can probably run jobs manually without a runner (like what I did with Docker Swarm before we had the runner working).

Yes, we are going to make a runner for k8s. There is some hacked-together runner at https://github.com/ipfs/testground/pull/217 that starts a `deployment` on a k8s cluster, but it needs a bit more work to get to the state of the Docker Swarm runner.

**Comment by nonsense** [comment]
@raulk @daviddias can I get a review on this one, so that future PRs are not enormous. I divided this out of the other PR so that we can review Terraform playbooks independently from Testground code.

File: infra/k8s/README-aws.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+terraform apply
+```
+
+7. Update your local .kube config and context
+
+```
+aws eks update-kubeconfig --name $CLUSTER_NAME
+```
+
+8. Edit outputs.tf, add admin users to the config (whoever is going to operate the AWS EKS cluster), and export Terraform outputs
```
Comment (raulk) [inline infra/k8s/README-aws.md]: i.e. replace the `$YOUR_USERNAME` placeholders

File: infra/k8s/aws/terraform/backends/example-backend.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+bucket = "terraform-backend-testground"
+key    = "$CLUSTER_NAME.tfstate"
```
Comment (raulk) [inline infra/k8s/aws/terraform/backends/example-backend.tf]:
```suggestion
key    = "<cluster_name>.tfstate"
```

File: infra/k8s/aws/terraform/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+data:
+  mapRoles: |
+    - rolearn: ${aws_iam_role.node-public.arn}
+      username: system:node:{{EC2PrivateDNSName}}
+      groups:
+        - system:bootstrappers
+        - system:nodes
+  mapUsers: |
+    - userarn: arn:aws:iam::909427826938:user/YOUR_USERNAME
+      username: $YOUR_USERNAME
```
Comment (raulk) [inline infra/k8s/aws/terraform/outputs.tf]:
```suggestion
      username: <your_aws_username>
```

File: infra/k8s/aws/terraform/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  namespace: kube-system
+data:
+  mapRoles: |
+    - rolearn: ${aws_iam_role.node-public.arn}
+      username: system:node:{{EC2PrivateDNSName}}
+      groups:
+        - system:bootstrappers
+        - system:nodes
+  mapUsers: |
+    - userarn: arn:aws:iam::909427826938:user/YOUR_USERNAME
```
Comment (raulk) [inline infra/k8s/aws/terraform/outputs.tf]:
```suggestion
    - userarn: <user_arn, e.g. arn:aws:iam::909427826938:user/YOUR_USERNAME, click on your user in https://console.aws.amazon.com/iam/home?#/users, and copy it>
```

File: infra/k8s/aws/terraform/terraform.tfvars-example [hunk 5]
```diff
@@ -0,0 +1,4 @@
+cluster-name   = "$CLUSTER_NAME"
```
Comment (raulk) [inline infra/k8s/aws/terraform/terraform.tfvars-example]:
```suggestion
cluster-name   = "<cluster_name>"
```

File: infra/k8s/aws/terraform/terraform.tfvars-example [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+cluster-region = "eu-central-1"
+key-name       = "$YOUR_KEY_NAME"
```
Comment (raulk) [inline infra/k8s/aws/terraform/terraform.tfvars-example]:
```suggestion
key-name       = "<key_name, doesn't need to match anything>"
```

File: infra/k8s/aws/terraform/terraform.tfvars-example [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+public-key     = "$PUBLIC_KEY"
```
Comment (raulk) [inline infra/k8s/aws/terraform/terraform.tfvars-example]:
```suggestion
public-key     = "<copy your pubkey here>"
```

**Pull request merged**